### PR TITLE
HPP and Payment Links add metadata limits

### PR DESCRIPTION
**Pull request by betty-ho-cko**
# Description of changes in PR

See Gateway limit here: https://checkout.slack.com/archives/C2ZJRL7ED/p1654069351980219
Added maxItems of 20 and maxLength of 255

## Checklist

- [ ] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [ ] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: abc_spec/components/schemas/HostedPayments/HostedPaymentsRequest.yaml [hunk 1]
```diff
@@ -189,8 +189,10 @@ properties:
   metadata:
     type: object
     title: The Metadata Schema
-    description: Allows you to store additional information about the transaction. This object only allows one level of depth, so cannot accept non-primitive data types such as objects or arrays.
+    description: Allows you to store additional information about the transaction. This object only allows one level of depth, so cannot accept non-primitive data types such as objects or arrays.
```
Comment (ben-ahmady-cko) [inline abc_spec/components/schemas/HostedPayments/HostedPaymentsRequest.yaml:192]:
```suggestion
    description: Allows you to store additional information about the transaction. This object only allows one level of depth, so cannot accept non-primitive data types such as objects or arrays.
```

**Review by ben-ahmady-cko — COMMENTED** [review]
Hey Betty this looks fine to me based on my understanding of the thread but would someone from GW be able to verify? It seems based on https://checkout.slack.com/archives/C2ZJRL7ED/p1654069351980219 that there isn't a consensus

**Comment by betty-ho-cko** [comment]
Thanks Ben. I'll follow up with Gateway to confirm and ask them to also raise their own PR

**Comment by betty-ho-cko** [comment]
Pending response from FS ticket: https://checkoutsupport.freshservice.com/helpdesk/tickets/145037

**Comment by betty-ho-cko** [comment]
@ben-ahmady-cko coming back to this as I forgot to follow up. Gateway ended up coming back to the question and provided the following response:

> We added a 255 character validation on the metadata fields last week and rolled that change back earlier today: https://checkout.atlassian.net/browse/GTW-7453

**Comment by betty-ho-cko** [comment]
No longer required

**Pull request closed without merging**